src/v/storage/types.h

```diff
@@ -38,6 +38,11 @@ constexpr model::record_batch_type ghost_record_batch_type
 using log_clock = ss::lowres_clock;
 using debug_sanitize_files = ss::bool_class<struct debug_sanitize_files_tag>;
 
+class stm_manager {
```

Comment:
add comments on this interface? docstring of how to use.

Reply:
+1